app/build.gradle

```diff
@@ -39,10 +41,12 @@ android {
             unitTests.includeAndroidResources = true
             unitTests.returnDefaultValues = true
         }
-        // defining the room.schemaLocation explicitly allows AutoMigrations
-        kapt {
-            arguments {
-                arg("room.schemaLocation", "$projectDir/schemas")
+
+        ksp {
+            // defining the room.schemaLocation explicitly allows AutoMigrations
+            room {
+                generateKotlin
+                schemaDirectory "$projectDir/schemas/"
```

Comment:
try:

```
ksp {
    arg("room.generateKotlin", "true")
    arg("room.schemaLocation", "$projectDir/schemas")
}
```

Reply:
Thanks a lot, it worked =)

Do you maybe also know where this needs to be resolved: 
`e: [ksp] Entities and POJOs must have a usable public constructor. You can have an empty constructor or a constructor whose parameters match the fields (by name and type). - android.database.Cursor`

Regrettably there are no further details, I tried around for a while and searched on the internet, but I couldn't find anything. Or do you maybe know any working reference implementation?

Reply:
I got nothing on this one unfortunately. `android.database.Cursor` is used in [ICalDatabaseDao](
https://github.com/TechbeeAT/jtxBoard/blob/develop/app/src/main/java/at/techbee/jtx/database/ICalDatabaseDao.kt#L735) and [SyncContentProvider](https://github.com/TechbeeAT/jtxBoard/blob/develop/app/src/main/java/at/techbee/jtx/SyncContentProvider.kt#L360).

maybe worth asking in the ksp repo if there are any known issues with `android.database.Cursor` or Simple/SupportSQLiteQuery: https://github.com/google/ksp/issues

and thank you for the awesome app. =)

Reply:
Thanks a lot =)
In fact, with your comment I reviewed it again and came to the right place, it works now =)